### PR TITLE
Feature/add GetOwnIdentityAttributes use case

### PR DESCRIPTION
**Pull request by Milena-Czierlinski**
I recommend to refactor attributes.test (and other test files, too) in another changeset and split them into smaller test files, such that it is easier to read and before/after statements can be used more conveniently.

### CODE REVIEW

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController } from "@nmshd/consumption";
+import { Inject } from "typescript-ioc";
+import { LocalAttributeDTO } from "../../../types";
+import { SchemaRepository, SchemaValidator, UseCase } from "../../common";
+import { AttributeMapper, GetAttributesRequestQuery, GetAttributesUseCase } from "../../consumption";
+import { flattenObject } from "../requests/flattenObject";
+
+export interface GetOwnIdentityAttributesRequest {
+    withoutSuccessor?: boolean;
```
Comment (slavistan) [inline src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:10]: How about `latestVersionOnly`? I find `withoutSuccessor` confusing as (at least to me) it suggests, that this use case _gets all own identity attributes without successor_, which is the exact opposite of what the implementation does.

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        const query: GetAttributesRequestQuery = {
+            shareInfo: "!"
+        };
+
+        if (typeof request.withoutSuccessor === "undefined" || request.withoutSuccessor) {
+            query["succeededBy"] = "!";
+        }
+
+        const flattenedQuery = flattenObject(query);
+        const dbQuery = GetAttributesUseCase.queryTranslator.parse(flattenedQuery);
```
Comment (slavistan) [inline src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:39]: Why not use a query object a la `{ $exists: false }` etc.? Is there a reason to use query strings?

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts (hunk 1 above)
Comment (stnmtz) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:10]: Har har, interesting - you could read it like that (when thinking about the successor property). We thought of "there is no successor of this attribute - thus return it" -> that is what the implementation does.

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts (hunk 2 above)
Comment (stnmtz) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:39]: You're right - if we do not have additional query parameters (which we do not have in this request it seems), than we should use the internal query syntax and don't use the flatten/queryTranslator case. I guess this has been copied from the GetAttributes use-case, where we have a customizable query?

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts (hunk 1 above)
Comment (Milena-Czierlinski) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:10]: I'm not set on this name. Any strong opinions?

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts (hunk 2 above)
Comment (Milena-Czierlinski) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:39]: It doesn't work using `queryTranslator`. However, I'm not completely sure, why we use `queryTranslator`. I just saw that it is used like this in all other use cases.

Comment (jkoenig134) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:39]: (or omit the translator completely :) )

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts (hunk 1 above)
Comment (jkoenig134) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:10]: I think `latestVersionsOnly` would be the least confusing (note the `s` in Versions!)

File: src/extensibility/facades/consumption/AttributesFacade.ts [hunk 3]
```diff
@@ -41,13 +41,15 @@ import {
     ValidateIQLQueryResponse,
     ValidateIQLQueryUseCase
 } from "../../../useCases";
+import { GetOwnIdentityAttributesRequest, GetOwnIdentityAttributesUseCase } from "../../../useCases/consumption/attributes/GetOwnIdentityAttributes";
```
Comment (jkoenig134) [inline src/extensibility/facades/consumption/AttributesFacade.ts:44]: missing export in `../../../useCases`

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController } from "@nmshd/consumption";
+import { Inject } from "typescript-ioc";
+import { LocalAttributeDTO } from "../../../types";
+import { SchemaRepository, SchemaValidator, UseCase } from "../../common";
+import { AttributeMapper, GetAttributesRequestQuery, GetAttributesUseCase } from "../../consumption";
+import { flattenObject } from "../requests/flattenObject";
```
Comment (jkoenig134) [inline src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:7]: this should be moved to another place (I know it's not new, but we should not access the "request space")

Comment (jkoenig134) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:7]: P.S.: the common directory would be a great place

File: src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts (hunk 1 above)
Comment (Milena-Czierlinski) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:10]: Can I warm you for `onlyLatestVersions`, too? :D

Comment (jkoenig134) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts:10]: sure

Comment (stnmtz) [reply src/useCases/consumption/attributes/GetOwnIdentityAttributes.ts]: I'd really love to veto just to see your faces, but go ahead ;)

**Review by jkoenig134 — DISMISSED** [review]

**Comment by jkoenig134** [comment]
please update with main (don't forget to check if the version is upgraded)

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**